dev/run-tests.py

```diff
@@ -0,0 +1,417 @@
+    run_cmd(["./dev/lint-python"])
+
+
+def exec_maven(mvn_args=[]):
+    """Will call Maven in the current directory with the list of mvn_args passed
+    in and returns the subprocess for any further processing"""
+
+    run_cmd(["./build/mvn"] + mvn_args)
+
+
+def exec_sbt(sbt_args=[]):
+    """Will call SBT in the current directory with the list of mvn_args passed
+    in and returns the subprocess for any further processing"""
+
+    sbt_cmd = ["./build/sbt"] + sbt_args
+
+    # NOTE: echo "q" is needed because sbt on encountering a build file
+    # with failure (either resolution or compilation) prompts the user for
+    # input either q, r, etc to quit or retry. This echo is there to make it
+    # not block.
+    echo_proc = subprocess.Popen(["echo", "\"q\n\""], stdout=subprocess.PIPE)
+    sbt_proc = subprocess.Popen(sbt_cmd,
+                                stdin=echo_proc.stdout,
+                                stdout=subprocess.PIPE)
+    echo_proc.wait()
+    for line in iter(sbt_proc.stdout.readline, ''):
+        if not SBT_OUTPUT_FILTER.match(line):
+            print line,    
+    retcode = sbt_proc.wait()
+
+    if retcode > 0:
+        exit_from_command_with_retcode(sbt_cmd, retcode)
+
+
+def build_apache_spark():
+    """Will first build Spark with Hive v0.12.0 to ensure the build is
+    successful and, after, will build Spark again against Hive v0.13.1 as the
+    tests are based off that"""
+
+    set_title_and_block("Building Spark", "BLOCK_BUILD")
+
+    sbt_maven_profile_args = os.environ.get(SBT_MAVEN_PROFILE_ARGS_ENV).split()
+    hive_profile_args = sbt_maven_profile_args + ["-Phive", 
+                                                  "-Phive-thriftserver"]
+    # set the default maven args
+    base_mvn_args = ["clean", "package", "-DskipTests"]
+    # set the necessary sbt goals
+    sbt_hive_goals = ["package", 
+                      "assembly/assembly", 
+                      "streaming-kafka-assembly/assembly"]
+
+    # Then build with default Hive version (0.13.1) because tests are based on
+    # this version
+    print "[info] Compile with Hive 0.13.1"
+    rm_r("lib_managed")
+    print "[info] Building Spark with these arguments:", 
+    print " ".join(hive_profile_args)
+
+    if AMPLAB_JENKINS_BUILD_TOOL == "maven":
+        exec_maven(hive_profile_args + base_mvn_args)
+    else:
+        exec_sbt(hive_profile_args + sbt_hive_goals)
+
+
+def detect_binary_inop_with_mima():
+    set_title_and_block("Detecting binary incompatibilities with MiMa",
+                        "BLOCK_MIMA")
+    run_cmd(["./dev/mima"])
+
+
+def run_scala_tests(test_suite=[]):
+    """Function to properly execute all tests pass in, as a list, from the
+    `determine_test_suite` function"""
+    set_title_and_block("Running Spark unit tests", "BLOCK_SPARK_UNIT_TESTS")
+
+    # ensure the test_suite is a set
+    if not isinstance(test_suite, set): 
```

Comment:
I think that you can just unconditionally wrap it into a set, since that should be idempotent:

```python
>>> set(set(set(set([1, 2, 3]))))
set([1, 2, 3])
```